spyder/widgets/comboboxes.py

```diff
@@ -279,10 +282,16 @@ def tab_complete(self):
         If there is a single option available one tab completes the option.
         """
         opts = self._complete_options()
+        if len(opts) == 0:
+            return
         if len(opts) == 1:
-            self.set_current_text(opts[0] + os.sep)
+            path = opts[0]
+            if osp.isdir(path):
+                path += os.sep
+            self.set_current_text(path)
             self.hide_completer()
         else:
+            self.set_current_text(os.path.commonprefix(opts))
```

Comment:
If these changes are necessary to list files in the working directory toolbar, then I think they should also be reverted.